### ISSUE TITLE
'RPC failed; curl 56 HTTP/2 stream 5 was reset' error when run command 'git submodule update --recursive --remote'

### ISSUE THREAD

**Issue by nbdinara** [git-for-windows/git#4469]
After the recent git update to version 2.41.0.windows.1 that was released on 6/1/2023, we started to see error when we run command 

```
git submodule update --recursive --remote
```

Error:
```
error: RPC failed; curl 56 HTTP/2 stream 5 was reset
fatal: the remote end hung up unexpectedly
fatal: clone of '<path to the repository>' into submodule path '<folder>' failed
```

### Setup

64-bit 
git version 2.41.0.windows.1

```
$ git --version --build-options

git version 2.41.0.windows.1
cpu: x86_64
built from commit: ff94e79c4724635915dbb3d4ba38f6bb91528260
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 11 Enterprise, version 22H2, 64-bit operating system
```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.22621.1848]

```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
> type "$env:USERPROFILE\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path: 
Default Branch Option:  
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Disabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

I can clone the main repository, but when I try to clone submodules using command git submodule update --recursive --remote I see the error listed above

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

PowerShell

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git submodule update --recursive --remote
```
 - What did you expect to occur after running these commands?

successful clone of submodules

 - What actually happened instead?

Error: 

error: RPC failed; curl 56 HTTP/2 stream 5 was reset
fatal: the remote end hung up unexpectedly
fatal: clone of '<path to the repository>' into submodule path '<folder>' failed

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Repository is private

**Comment by dscho** [comment]
Even if the repository is private, can you provide details as to the host on which this is happening? https://github.com/search?q=repo%3Acurl%2Fcurl+HTTP%2F2+stream+5+was+reset&type=issues lists a couple of tickets that raise the suspicion that this might be a server problem that has only been unhidden by recent libcurl updates.

**Comment by nbdinara** [comment]
> Even if the repository is private, can you provide details as to the host on which this is happening? https://github.com/search?q=repo%3Acurl%2Fcurl+HTTP%2F2+stream+5+was+reset&type=issues lists a couple of tickets that raise the suspicion that this might be a server problem that has only been unhidden by recent libcurl updates.

repository resides on Azure DevOps

**Comment by georgii-sosnovikov** [comment]
@nbdinara I reported it to the curl repository https://github.com/curl/curl/issues/11353, it looks like their [25/05/2023 fix](https://github.com/curl/curl/commit/5c58cb0212bcf63cce33a974906bf9905948b4bb) got into curl 8.1.2 and also into git 2.41.0.windows.1, but it doesn't work

**Comment by georgii-sosnovikov** [comment]
@dscho [the fix](https://github.com/curl/curl/commit/65937f0d638f152d955e327d0fdaf14d7c98cb29) for this was completed just 5 hours ago and the release of the new curl 8.2.0 is scheduled for July 19. Please update curl as soon as possible, because this is a problem for all git users.

**Comment by dscho** [comment]
> [the fix](https://github.com/curl/curl/commit/65937f0d638f152d955e327d0fdaf14d7c98cb29) for this was completed just 5 hours ago and the release of the new curl 8.2.0 is scheduled for July 19.

I looked into backporting it, but it seems that it relies on quite a few things that aren't in v8.1.2.

@georgii-sosnovikov would you mind testing the `libcurl-openssl-4.dll` from https://github.com/git-for-windows/MINGW-packages/actions/runs/5342706775 (once that's finished, this file should be contained in one of the uploaded artifacts and needs to replace your `C:\Program Files\Git\mingw64\bin\libcurl-openssl-4.dll`, at least temporarily)?

**Comment by georgii-sosnovikov** [comment]
@dscho This problem has not been reproduced for me, so there was no way to test this. By the way the CURL package has released a new version 8.2.0 with a fix for this bug and a bunch of others, would you consider updating it for the next release?

**Comment by rimrul** [comment]
> By the way the CURL package has released a new version 8.2.0 with a fix for this bug and a bunch of others, would you consider updating it for the next release?

Done (#4512)

**Comment by dscho** [comment]
@nbdinara please test the [latest snapshot](https://wingit.blob.core.windows.net/files/index.html), it comes with the fix, as will the imminent v2.42.0-rc0.
